File: langhuan-service/src/main/java/com/langhuan/utils/chatMemory/ChatMemoryUtils.java
package com.langhuan.utils.chatMemory;

import org.springframework.ai.chat.messages.Message;
import org.springframework.ai.chat.messages.MessageType;

import java.util.Map;

public class ChatMemoryUtils {
    public static Message createdMessage(String text, Map<String, Object> metadata, MessageType messageType) {
        return new Message() {
            @Override
            public String getText() {
                return text;
            }

            @Override
            public Map<String, Object> getMetadata() {
                return metadata;
            }

            @Override
            public MessageType getMessageType() {
                return messageType;
            }
        };
    }
}

File: langhuan-server/src/main/kotlin/com/langhuan/utils/chatMemory/ChatMemoryUtils.kt
package com.langhuan.utils.chatMemory

import org.springframework.ai.chat.messages.Message
import org.springframework.ai.chat.messages.MessageType

object ChatMemoryUtils {
    
    fun createdMessage(
        text: String,
        metadata: Map<String, Any>,
        messageType: MessageType
    ): Message {
        return object : Message {
            override fun getText(): String = text
            
            override fun getMetadata(): Map<String, Any> = metadata
            
            override fun getMessageType(): MessageType = messageType
        }
    }
}
